src/main/java/seedu/toluist/commons/util/StringUtil.java

```diff
@@ -55,4 +59,70 @@ public static String getDetails(Throwable t) {
     public static boolean isUnsignedInteger(String s) {
         return s != null && s.matches("^0*[1-9]\\d*$");
     }
+
+    /**
+     * Splits a string of indexes into a list.
+     *   Examples:
+     *   splitIndexes(" - 3",           8) -> [1, 2, 3]
+     *   splitIndexes(" 3 -",           8) -> [3, 4, 5, 6, 7, 8]
+     *   splitIndexes("3 4,  5",        8) -> [3, 4, 5]
+     *   splitIndexes("3 - 5",          8) -> [3, 4, 5]
+     *   splitIndexes("- 3, 5, 7 - 12", 8) -> [1, 2, 3, 5, 7, 8]
+     * This function is robust enough to handle excessive white spaces (" ") and commas (",").
+     * @param stringIndexes, the unprocessed string of indexes from user's input
+     * @param maxIndex, the maximum possible index number
+     * @return a list of integer containing all valid indexes
+     */
+    public static List<Integer> splitIndexes(String stringIndexes, int maxIndex) {
```

Comment:
Consider renaming it `getIndices` and handle the 1 index scenario here as well ('1' -> [1])